chisel/src/driver.rs

```diff
@@ -184,6 +205,278 @@ impl ChiselDriver {
         self.state = DriverState::Done(results);
         &self.state
     }
+
+    pub fn execute_module(
+        &mut self,
+        name: String,
+        module: ModuleConfig,
+        wasm: &mut Module,
+    ) -> Result<ModuleResult, DriverError> {
+        let result = match name.as_str() {
+            "checkfloat" => {
+                let checkfloat = CheckFloat::new();
+                let module_result = checkfloat.validate(wasm);
+                ModuleResult::Validator(name.clone(), module_result)
```

Comment:
Why not wrap these in `Ok()` and remove the match assignment?

Reply:
It'd be more concise but I think it looks a bit strange to have an `Ok()` wrapping an expression which could actually short-circuit and return `Err()` mid-evaluation